### DISCUSSION TITLE
Bootstrap error on Flux 0.13.4 for GitHub repo owned by organization

### DISCUSSION THREAD

**Discussion by nicogl** [fluxcd/flux2#1447] [Q&A]
I recently upgraded from Flux client 0.9.1 to 0.13.4, and now when I run the same flux bootstrap command I used to have, I get the following error:
► connecting to github.com
► reconciling repository permissions
✗ failed to grant "maintain" permissions to "TEAM_NAME": multiple errors occurred: 
- PUT https://api.github.com/orgs/MY_ORG/teams/TEAM_NAME/repos/MY_ORG/MY_REPO: 403 Rate Limit Exceeded []
- PUT https://api.github.com/orgs/MY_ORG/teams/TEAM_NAME/repos/MY_ORG/MY_REPO: 403 Rate Limit Exceeded []

[Documentation ](https://fluxcd.io/docs/installation/#github-and-github-enterprise) says: _"When you specify a list of teams, those teams will be granted maintainer access to the repository."_
It mentions the maintainer access permission but is not very specific on what it does or why it needs it. **What's the above HTTP PUT operation about?**

The GitHub personal access token I'm using is the same as before, having all "repo" permissions enabled.

After the message error from above, flux seems to install the components (controllers), but it remains on a loop and the command never exits. I get feedback messages about the deployments readiness but most of them say "deployment not ready":

► confirming components are healthy
✗ notification-controller: deployment not ready
✗ image-reflector-controller: deployment not ready
✗ image-automation-controller: deployment not ready
✗ source-controller: deployment not ready
✗ kustomize-controller: deployment not ready
✗ helm-controller: deployment not ready
✔ all components are healthy
✗ failed to grant permissions to team: reconciled with warning

Any idea? Could it be a bug in this version of the client?

**Comment by nicogl** [comment]
@stefanprodan @hiddeco 
Sorry to engage you directly guys but it's been some days since I posted this question and haven't received any answer so far.

Any comment/clue on what this may be? Could it be a potential bug in the 0.13.4 version?
Thanks in advance!

**Reply by nicogl** [reply]
@stefanprodan @hiddeco 
Excuse me again for my insistence on this topic. Do you know if this is/was effectively an issue? Any idea if latest version v0.16.0 may have the same behavior?
Thanks again

**Reply by stefanprodan** [reply]
To me this looks like a permissions issues, if you run bootstrap without specifying teams, does it work?

**Reply by nicogl** [reply]
This is interesting @stefanprodan !!
I ran the very same bootstrap command (**Flux v0.16.0**) without specifying _team_ and it seems to have run fine...

This is the original command with the _team_ parameter that failed with above error:

`flux bootstrap github --context=<MY_CONTEXT> --owner=<GITHUB_ORG> --repository=<REPO> --branch=<BRANCH> --team=<TEAM> --path=<PATH>  --components=source-controller,kustomize-controller,helm-controller,notification-controller --components-extra=image-reflector-controller,image-automation-controller`

Ran it without the _team_ parameter and worked fine.
Note that the original error says: 403 Rate Limit Exceeded   so doesn't seem to be a permission issue.

Originally I experienced the issue with Flux v0.13.4, but this time I tried with latest version 0.16.0.

FYI nothing changed at GitHub level (orgs, teams, permissions, users, config, etc) between bootstrap executions attempts.

Lastly, I noticed that the successful bootstrap (without _teams_ parameter) made only one commit to the repository, with message: **Add Flux v0.16.0 component manifests**. The second commit (with message: **Add flux v0.9.1 sync manifests**) that I used to have with version 0.9.1 wasn't done, but this might be the expected behavior. I brought it up just in case.

So with that being said, is it possible this is a bug?

**Reply by stefanprodan** [reply]
> So with that being said, is it possible this is a bug?

The 403 usually hides some permissions error. We do test every version of Flux against GitHub and our e2e tests are successfully adding teams to repos under an GH org. While this can be a bug, I'm surprised you're the only user so far having this issue. It's hard to tell where the bug is, if we can't replicate the error.

> The second commit (with message: Add flux v0.9.1 sync manifests) that I used to have with version 0.9.1 wasn't done, but this might be the expected behavior. 

That's normal, the sync manifests haven't changed between 0.9 and 0.16, only the components.

**Reply by nicogl** [reply]
Ok @stefanprodan ...
Let me ask you then, if I don't set the _teams_ parameter while bootstraping, what would I miss? I understand that the GitHub team wouldn't get the _maintainer_ access, but I saw that Flux is still able to commit changes to the repo... so what would be really missing without the maintainer role?

Finally, I agree that if this is a bug it is strange that no one else has raised it... however, it is really weird to me that running the very same bootstrap command with Flux v0.9.1 is fine and then immediately running with Flux v0.16.0 throws the error... in my humble opinion just doesn't make sense.

Thanks for everything!

**Comment by nicogl** [comment]
Ok @stefanprodan 
Let me ask you then, what would I miss if I don't set the _team_ parameter while bootstrapping? I understand that the _maintainer_ access wouldn't be granted to the GitHub Team, but so far I've seen Flux commiting to the CI repo without that role/permission. So do you see any potential issue without the _team_ parameter?

Finally, I agree that if this is a bug then it is very strange that no one else has also raised it. However, it is very weird to me that if I run the bootstrap command in Flux v0.9.1 it's fine and then swith to Flux v0.16.0 and run the very same command (without changes on GitHub) and it fails... In my humble opinion it just doesn't seem to make sense.

Thanks for everything!
